Sources/CryptoSwift/HashProtocol.swift

```diff
@@ -16,6 +16,10 @@ internal protocol HashProtocol: class {
 extension HashProtocol {
     
     func prepare(_ len:Int) -> Array<UInt8> {
+        return self.prepare(len, allowance: 8)
+    }
+    
+    func prepare(_ len:Int, allowance: Int) -> Array<UInt8> {
```

Comment:
it is ok name without specific implementation, that is bit padding now. It don't have to be like this. You've made a right point that it's general protocol and now I'm not sure if it should have shared prepare at all. 

for the purpose of this PR, this is how I see it:

```swift
func prepare(_ len:Int, allowance: Int = 0) -> Array<UInt8> {
}
```

overall, I'm very happy with the fix you provided.